sgx/report-cache/untrusted/src/lib.rs

```diff
@@ -208,39 +206,7 @@ impl<E: ReportableEnclave, R: RaClient> ReportCache<E, R> {
                 log::debug!(self.logger, "Enclave accepted report as valid...");
                 Ok(())
             }
-            Err(ReportableEnclaveError::AttestEnclave(AttestEnclaveError::Verify(
-                VerifierError::Verification(report_data),
-            ))) => {
-                // A verifier failed...
-                if let Some(platform_info_blob) = report_data.platform_info_blob.as_ref() {
-                    // IAS gave us a PIB
-                    log::debug!(
-                        self.logger,
-                        "IAS requested TCB update, attempting to update..."
-                    );
-                    QuotingEnclave::update_tcb(platform_info_blob)?;
-                    log::debug!(
-                        self.logger,
-                        "TCB update complete, restarting reporting process"
-                    );
-                    attestation_evidence = self.start_report_cache()?;
-                    log::debug!(
-                        self.logger,
-                        "Verifying attestation evidence with enclave (again)..."
-                    );
-                    self.enclave
-                        .verify_attestation_evidence(attestation_evidence.clone())?;
-                    log::debug!(self.logger, "Enclave accepted new report as valid...");
-                    Ok(())
-                } else {
-                    Err(Error::ReportableEnclave(
-                        ReportableEnclaveError::AttestEnclave(AttestEnclaveError::Verify(
-                            VerifierError::Verification(report_data),
-                        )),
-                    ))
-                }
-            }
-            Err(other) => Err(other.into()),
+            Err(e) => Err(e.into()),
```

Comment:
The newer DCAP attestation interface doesn't have this TCB update mechanism. It was the only place that the `VerificationReport` from the verification failure was looked at. Since it won't be used in the future just removed it instead of trying to re-implement it without the failed `VerificationReport`.

Reply:
How do you know that this isn't needed for normal operation while we are still not fully transitioned to DCAP?
I'm completely unfamiliar with this but there's probably a reason why this was here in the first place... and I have no idea what's going to break if its no longer there and we're still not switched to DCAP.

Reply:
Thank you!
That's a fair concern, and it's actually not that hard to maintain the same behavior.

I've added back the TCB update and retry behavior. Previously the `VerificationReportData` was decoded here, https://github.com/mobilecoinfoundation/mobilecoin/blob/6ccad2f67de22547c43398728a176a8770f82e41/attest/verifier/src/ias.rs#L225 and passed back in the error. Now the decoding happens in the error handler if it has the potential for a TCB update.